src/coreclr/interpreter/compiler.h

```diff
@@ -0,0 +1,226 @@
+        sVars[1] = sv2;
+        sVars[2] = sv3;
+    }
+};
+
+#define CALL_ARGS_SVAR  -2
+
+
+struct InterpBasicBlock
+{
+    int index;
+    int32_t ilOffset, nativeOffset;
+    int stackHeight;
+
+    InterpInst *pFirstIns, *pLastIns;
+    InterpBasicBlock *pNextBB;
+
+    int inCount;
+    InterpBasicBlock **ppInBBs;
+    int outCount;
+    InterpBasicBlock **ppOutBBs;
+
+};
+
+struct InterpVar
+{
+    CORINFO_CLASS_HANDLE clsHnd;
+    InterpType mt;
+    int indirects;
+    int offset;
+    int size;
+    // live_start and live_end are used by the offset allocator
+    int liveStart;
+    int liveEnd;
+};
+
+struct StackInfo
+{
+    StackType type;
+    CORINFO_CLASS_HANDLE clsHnd;
+    // Size that this value will occupy on the interpreter stack. It is a multiple
+    // of INTERP_STACK_SLOT_SIZE
+    int size;
+
+    // The var associated with the value of this stack entry. Every time we push on
+    // the stack a new var is created.
+    int var;
+};
+
+typedef class ICorJitInfo* COMP_HANDLE;
+
+class InterpCompiler
+{
+private:
+    CORINFO_METHOD_HANDLE m_methodHnd;
+    COMP_HANDLE m_compHnd;
+    CORINFO_METHOD_INFO* m_methodInfo;
+
+    int GenerateCode(CORINFO_METHOD_INFO* methodInfo);
+
+    void* AllocMethodData(size_t numBytes);
+    void* AllocMemPool(size_t numBytes);
+    void* AllocTemporary(size_t numBytes);
+    void* ReallocTemporary(void* ptr, size_t numBytes);
+    void  FreeTemporary(void* ptr);
+
+    // Instructions
+    InterpBasicBlock *m_pCBB, *m_pEntryBB;
+    InterpInst* m_pLastIns;
+
+    int32_t     GetInsLength(InterpInst *pIns);
+    bool        InsIsNop(InterpInst *pIns);
+    InterpInst* AddIns(int opcode);
+    InterpInst* NewIns(int opcode, int len);
+    InterpInst* AddInsExplicit(int opcode, int dataLen);
+    InterpInst* InsertInsBB(InterpBasicBlock *pBB, InterpInst *pPrevIns, int opcode);
+    InterpInst* InsertIns(InterpInst *pPrevIns, int opcode);
+    InterpInst* FirstIns(InterpBasicBlock *pBB);
+    InterpInst* NextIns(InterpInst *pIns);
+    InterpInst* PrevIns(InterpInst *pIns);
+    void        ClearIns(InterpInst *pIns);
+
+    void        ForEachInsSVar(InterpInst *ins, void *pData, void (InterpCompiler::*callback)(int*, void*));
+    void        ForEachInsVar(InterpInst *ins, void *pData, void (InterpCompiler::*callback)(int*, void*));
+
+    // Basic blocks
+    int m_BBCount = 0;
+    InterpBasicBlock**  m_ppOffsetToBB;
+
+    InterpBasicBlock*   AllocBB();
+    InterpBasicBlock*   GetBB(int32_t ilOffset);
+    void                LinkBBs(InterpBasicBlock *from, InterpBasicBlock *to);
+    void                UnlinkBBs(InterpBasicBlock *from, InterpBasicBlock *to);
+
+    // Vars
+    InterpVar *m_pVars = NULL;
+    int32_t m_varsSize = 0;
+    int32_t m_varsCapacity = 0;
+
+    int32_t CreateVarExplicit(InterpType mt, CORINFO_CLASS_HANDLE clsHnd, int size);
+
+    int32_t m_totalVarsStackSize = 0;
+    int32_t m_paramAreaOffset = 0;
+    void    AllocVarOffsetCB(int *pVar, void *pData);
```

Comment:
It's unclear from the name and signature of these methods what they do/how to use them, might be worth adding a comment later